packages/host/app/resources/matrix-profile.ts

```diff
@@ -0,0 +1,43 @@
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+import { restartableTask } from 'ember-concurrency';
+import { Resource } from 'ember-resources';
+
+import type MatrixService from '../services/matrix-service';
+
+export class MatrixProfileResource extends Resource<{}> {
+  @tracked loading: Promise<void> | undefined;
+
+  @tracked avatarUrl: string | undefined;
+  @tracked displayName: string | undefined;
+
+  @service private declare matrixService: MatrixService;
+
+  modify(_positional: never[], _named: never) {
+    this.loading = this.load.perform();
+  }
+
+  setDisplayName(newName: string) {
+    // FIXME is this a weird interface?
```

Comment:
I’m not sure a resource makes sense for this, the `displayname` always comes in a `m.room.member` event anyway, so does it need to be fetched with `getProfileInfo`? 🤔